CompletionStage results in NPE

[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"

## Environment Details
* Helidon Version: 2.2.2
* Helidon SE or Helidon MP: MP
* JDK version: 11
* OS: Windows
* Docker version (if applicable): N/A

----------

## Problem Description
If you use a `CompletionStage` as a response it results in a NPE:

```
Apr 04, 2021 7:09:57 PM io.helidon.webserver.RequestRouting$RoutedRequest defaultHandler
WARNING: Default error handler: Unhandled exception encountered.
java.util.concurrent.ExecutionException: Unhandled 'cause' of this exception encountered.
        at io.helidon.webserver.RequestRouting$RoutedRequest.defaultHandler(RequestRouting.java:397)
        at io.helidon.webserver.RequestRouting$RoutedRequest.nextNoCheck(RequestRouting.java:377)
        at io.helidon.webserver.RequestRouting$RoutedRequest.next(RequestRouting.java:420)
        at io.helidon.webserver.jersey.ResponseWriter.failure(ResponseWriter.java:172)
        at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:438)
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:263)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234)
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
        at io.helidon.webserver.jersey.JerseySupport$JerseyHandler.lambda$doAccept$3(JerseySupport.java:293)
        at io.helidon.common.context.Contexts.runInContext(Contexts.java:117)
        at io.helidon.common.context.ContextAwareExecutorImpl.lambda$wrap$5(ContextAwareExecutorImpl.java:154)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NullPointerException
        at org.glassfish.jersey.internal.PropertiesResolver$1.resolveProperty(PropertiesResolver.java:88)
        at org.glassfish.jersey.internal.PropertiesResolver$1.resolveProperty(PropertiesResolver.java:83)
        at org.glassfish.jersey.server.ContainerRequest.resolveProperty(ContainerRequest.java:292)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.unwrapInvocableResponseType(ResourceMethodInvoker.java:489)  
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.lambda$invoke$2(ResourceMethodInvoker.java:468)
        at org.glassfish.jersey.process.internal.Stages$LinkedStage.apply(Stages.java:284)
        at org.glassfish.jersey.process.internal.Stages.process(Stages.java:147)
        at org.glassfish.jersey.server.ServerRuntime$Responder.processResponse(ServerRuntime.java:371)
        at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:363)
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:258)
        ... 14 more
```

## Steps to reproduce
1. Use `helidon-mp-quickstart`
2. Create a resource that responds with `CompletionStage`:
```
	@GET
	@Path("foo")
	@Produces(MediaType.TEXT_PLAIN)
	public CompletionStage<String> foo() {
		return CompletableFuture.completedStage("bar");
	}
```

--- a/microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java
+++ b/microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java
@@ -1,0 +1,102 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *  
+ */
+
+package io.helidon.microprofile.server;
+
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class AsyncResourceTest extends Application {
+    private static Client client;
+
+    @BeforeAll
+    static void initClass() {
+        client = ClientBuilder.newClient();
+    }
+
+    @AfterAll
+    static void destroyClass() {
+        client.close();
+    }
+
+    @Test
+    void asyncResource() {
+        Server server = Server.builder()
+                .addApplication("/async-app", new TestApp())
+                .build();
+
+        server.start();
+
+        try {
+            WebTarget target = client.target("http://localhost:" + server.port());
+
+            String getResponse = target.path("/async-app/asyncGet").request().get(String.class);
+            assertThat(getResponse, is(equalTo("testResponseGet")));
+
+            String putResponse = target.path("/async-app/asyncPut").request().put(Entity.text(""), String.class);
+            assertThat(putResponse, is(equalTo("testResponsePut")));
+        } finally {
+            server.stop();
+        }
+    }
+
+    private final class TestApp extends Application {
+
+        @Override
+        public Set<Object> getSingletons() {
+            return Set.of(new TestResource());
+        }
+    }
+
+    @Path("/")
+    public static final class TestResource {
+
+        @GET
+        @Path("asyncGet")
+        @Produces(MediaType.TEXT_PLAIN)
+        public CompletionStage<String> foo() {
+            return CompletableFuture.completedStage("testResponseGet");
+        }
+        
+        @PUT
+        @Path("asyncPut")
+        public CompletionStage<Response> bar() {
+            return CompletableFuture.completedFuture(Response.ok().entity("testResponsePut").build());
+        }
+    }
+}

--- a/microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java
+++ b/microprofile/server/src/test/java/io/helidon/microprofile/server/AsyncResourceTest.java
@@ -12,12 +12,11 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- *  
+ *
  */
 
 package io.helidon.microprofile.server;
 
-import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 
@@ -25,66 +24,38 @@ import javax.ws.rs.GET;
 import javax.ws.rs.PUT;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
-import javax.ws.rs.client.Client;
-import javax.ws.rs.client.ClientBuilder;
 import javax.ws.rs.client.Entity;
 import javax.ws.rs.client.WebTarget;
-import javax.ws.rs.core.Application;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
+
+import io.helidon.microprofile.tests.junit5.AddBean;
+import io.helidon.microprofile.tests.junit5.HelidonTest;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.is;
 
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
-public class AsyncResourceTest extends Application {
-    private static Client client;
+@HelidonTest
+@AddBean(AsyncResourceTest.TestResource.class)
+public class AsyncResourceTest {
 
-    @BeforeAll
-    static void initClass() {
-        client = ClientBuilder.newClient();
-    }
-
-    @AfterAll
-    static void destroyClass() {
-        client.close();
+    @Test
+    void asyncResourceGet(WebTarget target) {
+        String getResponse = target.path("/asyncGet").request().get(String.class);
+        assertThat(getResponse, is(equalTo("testResponseGet")));
     }
 
     @Test
-    void asyncResource() {
-        Server server = Server.builder()
-                .addApplication("/async-app", new TestApp())
-                .build();
-
-        server.start();
-
-        try {
-            WebTarget target = client.target("http://localhost:" + server.port());
-
-            String getResponse = target.path("/async-app/asyncGet").request().get(String.class);
-            assertThat(getResponse, is(equalTo("testResponseGet")));
-
-            String putResponse = target.path("/async-app/asyncPut").request().put(Entity.text(""), String.class);
-            assertThat(putResponse, is(equalTo("testResponsePut")));
-        } finally {
-            server.stop();
-        }
-    }
-
-    private final class TestApp extends Application {
-
-        @Override
-        public Set<Object> getSingletons() {
-            return Set.of(new TestResource());
-        }
+    void asyncResourcePut(WebTarget target) {
+        String putResponse = target.path("/asyncPut").request().put(Entity.text(""), String.class);
+        assertThat(putResponse, is(equalTo("testResponsePut")));
     }
 
     @Path("/")
-    public static final class TestResource {
+    public static class TestResource {
 
         @GET
         @Path("asyncGet")
@@ -92,7 +63,7 @@ public class AsyncResourceTest extends Application {
         public CompletionStage<String> foo() {
             return CompletableFuture.completedStage("testResponseGet");
         }
-        
+
         @PUT
         @Path("asyncPut")
         public CompletionStage<Response> bar() {

--- a/webserver/jersey/src/main/java/io/helidon/webserver/jersey/JerseySupport.java
+++ b/webserver/jersey/src/main/java/io/helidon/webserver/jersey/JerseySupport.java
@@ -103,7 +103,7 @@ public class JerseySupport implements Service {
 
     private final ApplicationHandler appHandler;
     private final ExecutorService service;
-    private final JerseyHandler handler = new JerseyHandler();
+    private final JerseyHandler handler;
     private final HelidonJerseyContainer container;
     private final Thread serviceShutdownHook;
 
@@ -138,7 +138,7 @@ public class JerseySupport implements Service {
             // use the one provided
             builder.resourceConfig.register(AsyncExecutorProvider.create(builder.asyncExecutorService));
         }
-
+        this.handler = new JerseyHandler(builder.resourceConfig);
         this.appHandler = new ApplicationHandler(builder.resourceConfig, new ServerBinder(executorService));
         this.container = new HelidonJerseyContainer(appHandler, builder.resourceConfig);
 
@@ -235,6 +235,12 @@ public class JerseySupport implements Service {
      * for example.
      */
     private class JerseyHandler implements Handler {
+        
+        private final ResourceConfig resourceConfig;
+
+        public JerseyHandler(final ResourceConfig resourceConfig) {
+            this.resourceConfig = resourceConfig;
+        }
 
         @Override
         public void accept(ServerRequest req, ServerResponse res) {
@@ -261,7 +267,7 @@ public class JerseySupport implements Service {
                                                                    req.method().name(),
                                                                    new WebServerSecurityContext(),
                                                                    new MapPropertiesDelegate(),
-                                                                   null);
+                                                                    resourceConfig);
             // set headers
             req.headers().toMap().forEach(requestContext::headers);
 

--- a/webserver/jersey/src/main/java/io/helidon/webserver/jersey/JerseySupport.java
+++ b/webserver/jersey/src/main/java/io/helidon/webserver/jersey/JerseySupport.java
@@ -235,10 +235,10 @@ public class JerseySupport implements Service {
      * for example.
      */
     private class JerseyHandler implements Handler {
-        
+
         private final ResourceConfig resourceConfig;
 
-        public JerseyHandler(final ResourceConfig resourceConfig) {
+        JerseyHandler(final ResourceConfig resourceConfig) {
             this.resourceConfig = resourceConfig;
         }
 

